plugins/hls-tactics-plugin/src/Ide/Plugin/Tactic/Simplify.hs

```diff
@@ -0,0 +1,122 @@
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE PatternSynonyms     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE ViewPatterns        #-}
+
+module Ide.Plugin.Tactic.Simplify
+  ( simplify
+  ) where
+
+import Data.Data (Data)
+import Data.Generics (everywhere, somewhere, something, listify, extT, mkT, GenericT, mkQ)
+import Data.List.Extra (unsnoc)
+import Data.Maybe (isJust)
+import Data.Monoid (Endo (..))
+import Development.IDE.GHC.Compat
+import GHC.Exts (fromString)
+import GHC.SourceGen (var, op)
+import GHC.SourceGen.Expr (lambda)
+import Ide.Plugin.Tactic.CodeGen.Utils
+import Ide.Plugin.Tactic.GHC (fromPatCompatPs)
+
+
+------------------------------------------------------------------------------
+-- | A pattern over the otherwise (extremely) messy AST for lambdas.
+pattern Lambda :: [Pat GhcPs] -> HsExpr GhcPs -> HsExpr GhcPs
```

Comment:
Maybe NonEmpty list could be used here? Pretty minor though

Reply:
It's nice if this were a unipattern, but bidirectionally it just pushes the logic onto all callers. I'd prefer not to, unless you care strongly.

Reply:
Tho I guess as it stands, this will fail:

```
do
  Lambda [] bar' <- pure $ Lambda [] bar
```

Reply:
Your call.

Reply:
Ehh, I don't care enough. Feel free to push back if you do!